lib/ddtrace/correlation.rb

```diff
@@ -6,21 +6,24 @@ module Datadog
   # e.g. Retrieve a correlation to the current trace for logging, etc.
   module Correlation
     # Struct representing correlation
-    Identifier = Struct.new(:trace_id, :span_id, :env, :version) do
+    Identifier = Struct.new(:trace_id, :span_id, :env, :service, :version) do
       def initialize(*args)
         super
         self.trace_id = trace_id || 0
         self.span_id = span_id || 0
         self.env = env || Datadog.configuration.env
+        self.service = service || Datadog.configuration.service
         self.version = version || Datadog.configuration.version
       end
 
       def to_s
-        str =  "#{Ext::Correlation::ATTR_TRACE_ID}=#{trace_id}"
+        str = ''
+        str += "#{Ext::Correlation::ATTR_ENV}=#{env}" unless env.nil?
```

Comment:
would `["env=env", "version=version", ...].join(" ")` be easier to read/manage?

Reply:
Hmmmm, yes array joining would be cleaner. :P